### PR TITLE
Refine components to improve physics performance

### PR DESCRIPTION
**Pull request by WenhaiLin**

### CODE REVIEW

**Comment by WenhaiLin** [comment]
@minggo Please review this PR

File: cocos/2d/CCNode.h [hunk 1]
```diff
@@ -1862,6 +1847,19 @@ class CC_DLL Node : public Ref
...
+    PhysicsBody* _physicsBody;
+public:
+    void setPhysicsBody(Component* physicsBody) 
+    { 
+        addComponent(physicsBody);
+    }
+    PhysicsBody* getPhysicsBody() const { return _physicsBody; }
+
+    friend class PhysicsBody;
+#endif
```
Comment (minggo) [inline cocos/2d/CCNode.h:1861]: Please don't keep these functions. It is strange that Node can get physicsbody from component and from `getPhysicsBody()`.

File: cocos/editor-support/cocostudio/CCComAudio.cpp [hunk 2]
```diff
@@ -29,14 +29,17 @@ THE SOFTWARE.
 namespace cocostudio {
 
 IMPLEMENT_CLASS_COMPONENT_INFO(ComAudio)
-ComAudio::ComAudio(void)
+
+const std::string ComAudio::COMPONENT_NAME = "CCComAudio";
```
Comment (minggo) [inline cocos/editor-support/cocostudio/CCComAudio.cpp]: Why not add it in `Component`?

File: cocos/2d/CCNode.h (hunk 1 above)
Comment (minggo) [inline cocos/2d/CCNode.h:1861]: Ok, it can improve performance and keep compatibility.

**Comment by minggo** [comment]
This PR looks good to me.

File: cocos/2d/CCNode.h (hunk 1 above)
Comment (WenhaiLin) [inline cocos/2d/CCNode.h:1861]: I think backward compatibility is important unless we redesigned for better physical.

**Pull request merged**